databricks/koalas/frame.py

```diff
@@ -164,7 +164,7 @@ def _init_from_spark(self, sdf, metadata=None):
         else:
             self._metadata = metadata
 
-    def _reduce_for_stat_function(self, sfun):
+    def _reduce_for_stat_function(self, sfun, numeric_only=None):
```

Comment:
should this just default to True rather than None? Then you can remove that line below.

Reply:
I agree that it would make more sense, but the signature for all stats function on pandas have `None` as a default value for `numeric_only`. So I could replicate the conversion to a Boolean value inside all methods that call `_reduce_for_stat_function`, or keep the conversion inside `_reduce_for_stat_function`. I'd prefer keeping it inside to remove duplicate code, but I am open to suggestions :-) wdyt ?

Reply:
@rxin Let me know if you want me to change anything :-)

Reply:
I would also argue against `None` as the default, especially since pandas' way of [handling `None`](https://github.com/pandas-dev/pandas/blob/master/pandas/core/frame.py#L7497) seems rather unintuitive to me.

When for example running `pd.DataFrame({'A': [1, 2, 3], 'B': ['a', 'b', 'c']}).max()`, we get
```
A    3
B    c
dtype: object
```
as a result which shows that by default `numeric_only` should be `False` (this can also be verified by looking at the code linked to above).

However, `numeric_only=True` should probably still be used for some aggregations since for example the mean and median of strings are hard to make sense of. In fact, pandas also doesn't output any value for column *B* when calling `.mean()` on the DataFrame above.

Reply:
Seems fair. So if I recap, for some aggregations method, the default value should be `False`, for some it should be `True` (which is fine i guess). Consequently, there is no reasonable default value for `_reduce_for_stat_function`, which means the most reasonable choice here is probably to simply remove the default value, and make sure we pass down a value for `numeric_only` every time we call it.

Reply:
I agree - not having a difficult value seems better.

Reply:
I'd argue that the default value should be `False` since ignoring non-numerical values should only happen if explicitly stated. `numeric_only=False` however, uses all columns of the DataFrame which I'd say is what the user expects without specifying anything.

Before making any changes to your code, I would probably wait for an opinion from @rxin though.

Reply:
Sorry @floscha I don't have a strong opinion here. I think we should just follow your suggestion :)